docs/pages/pmd/rules/java/performance.md

```diff
@@ -109,6 +109,68 @@ public class Test {
 <rule ref="category/java/performance.xml/AvoidArrayLoops" />
 ```
 
+## AvoidFileStream
+
+**Since:** PMD 6.0.0
+
+**Priority:** High (1)
+
+Avoid instantiating FileInputStream, FileOutputStream, FileReader, or FileWriter.
+- Use Files.newInputStream(Paths.get(fileName)) in place of new FileInputStream(fileName).
+- Use Files.newOutputStream(Paths.get(fileName)) instead of new FileOutputStream(fileName).
+- Use Files.newBufferedReader(Paths.get(fileName)) instead of new FileReader(fileName).
+- Use Files.newBufferedWriter(Paths.get(fileName)) instead of new FileWriter(fileName).
+
+The FileInputStream and FileOutputStream classes contains a finalizer method which will cause garbage collection pauses.
+See [JDK-8080225](https://bugs.openjdk.java.net/browse/JDK-8080225) for details.
+
+The FileReader and FileWriter constructors instantiate FileInputStream and FileOutputStream,
+again causing garbage collection issues while finalizer methods are called.
```

Comment:
did you manually edit this file to add these last 2 paragraphs?

These files are automatically regenerated, so any edits will be lost. This info (very important and accurate) should be part of the rule description in the XML. If you can't do it, I'll edit this before merging, probably tomorrow.

Reply:
I did manually edit.  I can update tomorrow.